### PR TITLE
chatterino2: use wrapQtAppsHook, unstable-2019-05-11 -> v2.1.7, adopt as maintainer

### PR DESCRIPTION
**Pull request by ldesgoui**
###### Motivation for this change
The current `chatterino2` derivation does not provide a functioning client because of the breaking changes caused by the addition of `wrapQtAppsHook`.
I have also updated the sources which were out of date as the project moved to a Github organisation.
I have also added myself as a maintainer.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)) [NA]
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date [NA]
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/chatterino2/default.nix [hunk 1]
```diff
@@ -1,28 +1,42 @@
...
-  version = "unstable-2019-05-11";
+  version = "v2.1.7";
+
   src = fetchFromGitHub {
-    owner = "fourtf";
+    owner = "Chatterino";
     repo = pname;
-    rev = "8c46cbf571dc8fd77287bf3186445ff52b1d1aaf";
-    sha256 = "0i2385hamhd9i7jdy906cfrd81cybw524j92l87c8pzrkxphignk";
+    rev = version;
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/chatterino2/default.nix]:
```suggestion
    rev = "$v${version};
```

please no letters in `version`.

File: pkgs/applications/networking/instant-messengers/chatterino2/default.nix [hunk 2]
```diff
@@ -1,28 +1,42 @@
...
+    sha256 = "0bbdzainfa7hlz5p0jfq4y04i3wix7z3i6w193906bi4gr9wilpg";
     fetchSubmodules = true;
   };
-  nativeBuildInputs = [ qmake pkgconfig ];
-  buildInputs = [ qtbase qtsvg qtmultimedia boost openssl ];
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/chatterino2/default.nix]:
```suggestion
```

File: pkgs/applications/networking/instant-messengers/chatterino2/default.nix [hunk 3]
```diff
@@ -1,28 +1,42 @@
-{ stdenv, pkgconfig, fetchFromGitHub, qtbase, qtsvg, qtmultimedia, qmake, boost, openssl }:
+{ stdenv, pkgconfig, fetchFromGitHub, qtbase, qtsvg, qtmultimedia, qmake, boost, openssl, wrapQtAppsHook }:
 
 stdenv.mkDerivation rec {
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/chatterino2/default.nix]:
```suggestion
mkDerivation rec {
```

File: pkgs/applications/networking/instant-messengers/chatterino2/default.nix [hunk 4]
```diff
@@ -1,28 +1,42 @@
-{ stdenv, pkgconfig, fetchFromGitHub, qtbase, qtsvg, qtmultimedia, qmake, boost, openssl }:
+{ stdenv, pkgconfig, fetchFromGitHub, qtbase, qtsvg, qtmultimedia, qmake, boost, openssl, wrapQtAppsHook }:
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/chatterino2/default.nix]:
```suggestion
{ stdenv, pkgconfig, fetchFromGitHub, qtbase, qtsvg, qtmultimedia, qmake, boost, openssl, mkDerivation }:
```

**Review by worldofpeace — CHANGES_REQUESTED** [review]
We should use the qt specific deriver `mkDerivation` which automatically adds `wrapQtAppsHook` etc.
> https://nixos.org/nixpkgs/manual/#sec-language-qt

**Comment by worldofpeace** [comment]
@ldesgoui Your commit message `chatterino2: bump version from unstable-2019-05-11 to v2.1.7` should be `chatterino2: 2019-05-11 -> 2.1.7` as per the commit guidelines.

**Comment by ldesgoui** [comment]
I addressed requests and force-pushed changes, thank you for taking a look @worldofpeace

**Pull request closed without merging**